juju/juju
# Report whether a model is a controller model

## Description of change

This PR adds the ability to identify whether a model is a controller model from the CLI and the GUI. 

From the [bug report](https://bugs.launchpad.net/juju/+bug/1803247)

> Because the controller model is "special" the GUI blocks certain actions in it, like the deletion action.
>
> The issue is that we can no longer determine if a model is a controller model as the `model-uuid` and the `controller-uuid` no longer match and the name "controller" can be changed.
>
> There doesn't appear to be any information in the `ModelInfo` call that can be used to determine if a model is a controller model. An 'is-controller' key would be ideal.

## QA steps

Step 0: Bootstrap and deploy a bundle:

- `juju bootstrap`
- `juju deploy wiki-simple`

Step 1: Verify that `show-model controller` includes `is-controller: true`. 

```plain
$ juju show-model controller
controller:
  name: admin/controller
  short-name: controller
  ...
  is-controller: true
  ...
``` 

Step 2: Verify that `show-model default` includes `is-controller: false`. 

```plain
$ juju show-model default
controller:
  name: admin/default
  short-name: default
  ...
  is-controller: false
  ...
``` 

## Documentation changes

todo

## Bug reference

lp: https://bugs.launchpad.net/juju/+bug/1803247
